hca/full_upload.py

```diff
@@ -0,0 +1,177 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import os
+import pprint
+import sys
+import uuid
+
+from io import open
+
+from .constants import Constants
+from .upload_to_cloud import upload_to_cloud
+
+
+def eprint(*args, **kwargs):
+    """Print to stderr."""
+    print(*args, file=sys.stderr, **kwargs)
+
+
+class FullUpload:
+    """Functions needed to fully add this functionality to the command line parser."""
+
+    CONSOLE_ARGUMENT = "upload"
+    FILE_OR_DIR_ARGNAME = "file_or_dir"
+    CREATOR_ID_ENVIRONMENT_VARIABLE = "creator_uid"
+    REPLICA_ARGNAME = "--replica"
+
+    @classmethod
+    def add_parser(cls, subparsers):
+        """Call from parser.py to create the parser."""
+        subparser = subparsers.add_parser(
+            cls.CONSOLE_ARGUMENT,
+            help="Upload a file or directory to the cloud, register each of the files, and bundle them together."
+        )
+
+        subparser.add_argument(
+            cls.FILE_OR_DIR_ARGNAME,
+            nargs="+",
+            help="Relative or direct path to folder with all bundle files."
+        )
+
+        subparser.add_argument(
+            cls.REPLICA_ARGNAME,
+            help="Which cloud to upload to first. One of 'aws', 'gc', or 'azure'.",
+            default="aws"
+        )
+        subparser.add_argument(
+            "--staging-bucket",
+            help="Bucket within replica to upload to.",
+            default="hca-dcp-staging-test")
```

Comment:
not sure you want to have a default value since most users would not be able to take advantage of it.